src/main/java/com/hms_networks/sc/cumulocity/data/CConnectorChildUpdate.java

```diff
@@ -0,0 +1,270 @@
+   * child ID provided.
+   *
+   * @param childId the id of the child device
+   * @since 1.0.0
+   */
+  public CConnectorChildUpdate(String childId) {
+    this.childDeviceId = childId;
+    readHardwareTags();
+    readFirmwareTags();
+  }
+
+  /**
+   * Try to read the hardware specific c8y tags.
+   *
+   * @since 1.0.0
+   */
+  private void readHardwareTags() {
+    hardwareModel = tryGetMetaDataTag(TAG_POSTFIX_HARDWARE_MODEL);
+    hardwareRevision = tryGetMetaDataTag(TAG_POSTFIX_HARDWARE_REVISION);
+    hardwareSerialNum = tryGetMetaDataTag(TAG_POSTFIX_HARDWARE_SERIAL_NUM);
+  }
+
+  /**
+   * Try to read the firmware specific c8y tags.
+   *
+   * @since 1.0.0
+   */
+  private void readFirmwareTags() {
+    firmwareName = tryGetMetaDataTag(TAG_POSTFIX_FIRMWARE_NAME);
+    firmwareVersion = tryGetMetaDataTag(TAG_POSTFIX_FIRMWARE_VERSION);
+    firmwareUrl = tryGetMetaDataTag(TAG_POSTFIX_FIRMWARE_URL);
+  }
+
+  /**
+   * Try to read the tag value for the given postfix. If the tag does exist, then return the value
+   * as a string.
+   *
+   * @param postfix the specific tag ending to read
+   * @return null, if tag could not be read. Otherwise, the value of the tag as a string.
+   * @since 1.0.0
+   */
+  private String tryGetMetaDataTag(String postfix) {
+    try {
+      TagControl tc = new TagControl(childDeviceId + "/" + TAG_METADATA_TOKEN + "/" + postfix);
+      numReadTags++;
+      return tc.getTagValueAsString();
+      // if the tag does not exist, it is not an EWException that is thrown as JavaDocs suggests, so
+      // here we catch all
+    } catch (Exception e) {
+      Logger.LOG_CRITICAL(
+          "Warning, did not find child metadata tag: "
+              + childDeviceId
+              + "/"
+              + TAG_METADATA_TOKEN
+              + "/"
+              + postfix);
+      Logger.LOG_EXCEPTION(e);
+    }
+    return null;
+  }
+
+  /**
+   * Get the JSON line fragment for the key and value. If the value is null, then an empty string is
+   * returned.
+   *
+   * @param key the key for the JSON object
+   * @param value the value for the JSON object
+   * @return string containing the JSON line fragment
+   * @since 1.0.0
+   */
+  private String getJsonLine(String key, String value) {
+    if (value != null) {
+      return "\"" + key + "\": \"" + value + "\"";
+    }
+    return "";
+  }
+
+  /**
+   * Helper method to return JSON when lines potentially could be empty strings. Handles where
+   * commas might be. C8y_Firmware and c8y_Hardware objects are expected to have 3 lines.
+   *
+   * @param key the key for the JSON object
+   * @param line1 the first line of the JSON object
+   * @param line2 the second line of the JSON object
+   * @param line3 the third line of the JSON object
+   * @return string containing the JSON object
+   * @since 1.0.0
+   */
+  private String getJsonThreeLineObject(String key, String line1, String line2, String line3) {
```

Comment:
Is there any reason for building JSON manually in this method and this class? We have the `JSONObject` class, which can be used to create JSON and JSON strings.

Reply:
In benchmarking, it was faster. Not sure about memory